src/Core/OntologyAnnotation.fs

```diff
@@ -131,31 +131,28 @@ type OntologyAnnotation(?name,?tsr,?tan, ?comments) =
         sb.Append("}") |> ignore
         sb.ToString()
 
-    override this.Equals other =
-        match other with
-        | :? OntologyAnnotation as oa -> (this :> System.IEquatable<_>).Equals oa
-        | :? string as s ->           
-            this.NameText = s
-            || 
-            this.TermAccessionShort = s
-            ||
-            this.TermAccessionOntobeeUrl = s
-        | _ -> false
-
-    override this.GetHashCode () = (this.NameText+this.TermAccessionShort).GetHashCode()
-
-    interface System.IEquatable<OntologyAnnotation> with
-        member this.Equals other =
-            if this.TermAccessionNumber.IsSome && other.TermAccessionNumber.IsSome then
-                other.TermAccessionShort = this.TermAccessionShort
-                ||
-                other.TermAccessionOntobeeUrl = this.TermAccessionOntobeeUrl
-            elif this.Name.IsSome && other.Name.IsSome then
-                other.NameText = this.NameText
-            elif this.TermAccessionNumber.IsNone && other.TermAccessionNumber.IsNone && this.Name.IsNone && other.Name.IsNone then
-                true
-            else 
-                false
+    override this.GetHashCode() = 
```

Comment:
How does this hash behave in cases, where the underlying Term is the same but once stored as obo url and once as short tan?

Reply:
taninfo uses information after regex parsing. So everything we support with our regex works fine. I also added some tests for this.